DOCS.md

```diff
@@ -0,0 +1,66 @@
+# Documentation
+
+This document intends to provide information on how to get the Vela application running locally.
+
+For more information, please see our [installation docs](https://go-vela.github.io/docs/install/).
+
+## Prerequisites
+
+* [Docker](https://docs.docker.com/install/) - building block for local development
+* [Docker Compose](https://docs.docker.com/compose/install/) - start up local development
+* [Github OAuth Client](https://developer.github.com/apps/building-oauth-apps/creating-an-oauth-app/) - building block for local Development
+* [Golang](https://golang.org/dl/) - for source code and [dependency management](https://github.com/golang/go/wiki/Modules)
+* [Make](https://www.gnu.org/software/make/) - start up local development
+
+## Setup
+
+* Clone this repository to your workstation:
+
+```bash
+# clone the project
+git clone git@github.com:go-vela/server.git $HOME/go-vela/server
+```
+
+* Navigate to the repository code:
+
+```bash
+# change into the cloned project directory
+cd $HOME/go-vela/server
+```
+
+* Create an [OAuth App](https://developer.github.com/apps/building-oauth-apps/creating-an-oauth-app/) and obtain secrets for local development:
+  * `Application name` = `Vela - local` (name of the OAuth application shouldn't matter)
+  * `Homepage URL` = `http://localhost:8080` (base URL of the server)
+  * `Authorization callback URL` = `http://localhost:8888/account/authenticate` (authenticate endpoint of the base URL of the UI)
+
+**NOTE: This will work for GitHub or GitHub Enterprise.**
+
+* Add OAuth client secrets to a local `secrets.env` file:
+
+```bash
+# add Github Client ID to local secrets file for `docker-compose`
+echo "VELA_SOURCE_CLIENT=<Github OAuth Client ID>" >> secrets.env
+
+# add Github Client Secret to local secrets file for `docker-compose`
+echo "VELA_SOURCE_SECRET=<Github OAuth Client Secret>" >> secrets.env
+```
+
+## Start
+
+**NOTE: Please review the [setup section](#setup) before starting.
+
+* Navigate to the repository code:
+
+```bash
+# change into the cloned project directory
+cd $HOME/go-vela/server
+```
+
+* Run the repository code:
+
+```bash
+# execute the `up` target with `make`
+make up
+```
+
```

Comment:
Any thoughts on adding notes about repo activation & faking out webhooks sent by GitHub?

Reply:
In regards to repo activation, would providing a link to our external docs on the topic suffice?

https://go-vela.github.io/docs/usage/getting-started/enable_repo/

As far as faking out webhooks go, what content would you like to see?

I think it's going to be slightly more challenging to accurately document that process since it requires a more in-depth knowledge of how webhooks work as well as the Vela system in general.

I'd almost rather focus efforts on writing a script that creates the webhook payload for the user 🤷

Reply:
I agree that documenting the process is somewhat challenging due to the expected knowledge. However, I think we could at least briefly mention the following:

- Process to update postgres repos table with the proper `link` and `clone` columns
    
   ```sh
  psql -U vela -h localhost
  UPDATE repos SET link = 'https://git.company.com/<org>/<repo>', clone = 'https://git.company.com/<org>/<repo>.git' WHERE id = <id>;
   ```
- Talk through the process of obtaining a real payload from `https://git.company.com/<org>/<repo>settings/hooks/<id> ` and sending that to the with curl, insomnia, etc. to generate a build
  - Providing a script to create some of that data would also work

Reply:
I'm happy to say that the first bullet point **shouldn't** be necessary very soon! 🎉 

Shoutout to @wass3r for https://github.com/go-vela/server/pull/200

The important part of that PR is Vela will now be querying GitHub for the repo information when creating a repo so the `link` and `clone` fields will be populated from that call rather than the way we were doing it previously.

In regards to the second bullet point, would you also expect to see docs instructing the user that the real payload needs to be from a commit with a valid Vela pipeline (`.vela.yml`/`.vela.yaml`)?

If so, then would you also expect to see documentation on how to create/setup a valid Vela pipeline?